File: app/src/main/java/com/islam/basepropject/project_base/common/ui/settings/SettingsFragment.java
package com.islam.basepropject.project_base.common.ui.settings;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.preference.ListPreference;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;

import com.islam.basepropject.R;
import com.islam.basepropject.project_base.base.activities.BaseActivity;

public class SettingsFragment extends PreferenceFragmentCompat {

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        ((BaseActivity)getActivity()).setToolbarTitle(R.string.settings);
    }

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        setPreferencesFromResource(R.xml.settings, rootKey);
        ListPreference listPreference = findPreference("language");
        listPreference.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
            @Override
            public boolean onPreferenceChange(Preference preference, Object newValue) {

                getActivity().recreate();
                return true;
            }
        });
        listPreference.setSummary(listPreference.getEntry());

    }
}

File: app/src/main/java/com/islam/basepropject/project_base/common/ui/settings/SettingsFragment.kt
package com.islam.basepropject.project_base.common.ui.settings

import android.os.Bundle
import androidx.preference.ListPreference
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat

import com.islam.basepropject.R
import com.islam.basepropject.project_base.base.activities.BaseActivity

class SettingsFragment : PreferenceFragmentCompat(),TagedFragment {

    override val fragmentTag: String
        get() = "SettingsFragment"

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        (activity as BaseActivity).setToolbarTitle(R.string.settings)
    }

    override fun onCreatePreferences(state: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.settings, rootKey)
        val listPreference = findPreference<ListPreference>("language")
        listPreference?.onPreferenceChangeListener = Preference.OnPreferenceChangeListener { preference, newValue ->
            activity!!.recreate()
            true
        }
        listPreference?.summary = listPreference?.entry

    }
}
